bdd-for-c.h

```diff
@@ -496,7 +536,7 @@ for(\
             )\
         )\
     );\
-    ++__bdd_index__\
+    __bdd_index__ = (char*)__bdd_index__ + 1\
```

Comment:
Why a cast is necessary here?

Reply:
Arithmetic on void* is a [nonstandard GCC extension][1] -- fails with error [C2036][2] in VC.

[1]: https://stackoverflow.com/questions/37460579/error-c2036-void-unknown-size
[2]: https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/compiler-error-c2036?view=vs-2019